File: clients/js/src/requests/trainer_names.ts
import { BaseRequest } from './base';

export type TrainerNameRequest = BaseRequest<TrainerNameRequestType, TrainerNameRequestAttributes>;

enum TrainerNameRequestType {
    trainer_names = 'trainer_names',
}

export interface TrainerNameRequestAttributes {
    name: string;
}

export const TrainerNameRequestDefault = (attributes: TrainerNameRequestAttributes): TrainerNameRequest => {
    return {
        data: {
            type: TrainerNameRequestType.trainer_names,
            attributes,
        },
    };
};

File: pkmnapi-api/src/requests/trainer_names.rs
use rocket_okapi::JsonSchema;
use serde::Deserialize;

use crate::requests::base::BaseRequest;

pub type TrainerNameRequest = BaseRequest<TrainerNameRequestType, TrainerNameRequestAttributes>;

impl TrainerNameRequest {
    pub fn get_name(&self) -> &String {
        &self.data.attributes.name
    }
}

#[derive(Debug, Deserialize, JsonSchema)]
#[allow(non_camel_case_types)]
pub enum TrainerNameRequestType {
    trainer_names,
}

#[derive(Debug, Deserialize, JsonSchema)]
pub struct TrainerNameRequestAttributes {
    pub name: String,
}
